### DISCUSSION TITLE
Deployment of projects of the new versions of Scully in vercel

### DISCUSSION THREAD

**Discussion by carlos-burelo** [scullyio/scully#1439] [Q&A]
Scully is currently supported by Vercel ?, I've been trying for weeks to display a website in vercel, but it can't seem to run puppeteer, specifically the browser.
I'm use

```json
{
"scripts":{
    "build":"ng build && scully",
  }
}
"dependencies":{
    "@angular/animations": "~12.2.0",
    "@angular/common": "~12.2.0",
    "@angular/compiler": "~12.2.0",
    "@angular/core": "~12.2.0",
    "@angular/forms": "~12.2.0",
    "@angular/platform-browser": "~12.2.0",
    "@angular/platform-browser-dynamic": "~12.2.0",
    "@angular/router": "^12.2.6",
    "@scullyio/init": "1.1.4",
    "@scullyio/ng-lib": "^1.0.0",
    "@scullyio/scully": "^1.0.0",
    "@scullyio/scully-plugin-base-href-rewrite": "^1.1.1",
}
``` 

and return this error

```bash
18:26:26.263
Puppeteer cannot find or launch the browser. (by default chrome)
18:26:26.263
 Try adding 'puppeteerLaunchOptions: {executablePath: CHROMIUM_PATH}'
18:26:26.263
 to your scully.*.config.ts file.
18:26:26.263
Also, this might happen because the default timeout (60 seconds) is to short on this system
18:26:26.263
this can be fixed by adding the --serverTimeout=x cmd line option.
18:26:26.263
   (where x = the new timeout in milliseconds)
18:26:26.263
When this happens in CI/CD you can find some additional information here:
18:26:26.263
https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md
```

**Comment by SanderElias** [comment]
There are a couple of things that can (and do) go haywire in CI/CD scenarios.
1. chrome isn't installed. 
     This should be handled by the puppeteer installation, but sometimes it doesn't work. The link in the error-message goes to a page about puppeteer deploy issues. Your answer is probably in there. 
     Also, you might need to use a different base-container that already includes Chrome, then you need to do the setting. When you do that, you need to teach Scully where your Chrome is installed, you can do that with environment settings, and with the setting in the setting, I will list below my personal environment settings for this.
     Then there are some settings related to docker (search for Docker in our issues). For some of the CI/CD systems those will come into play too.
2. timeouts
   The thing is, that some of the CI/CD environments are a bit slow with opening up ports for local use. When this is the case, you can increase the wait time. (this is the time Scully will wait and try to reach the server where the dist files are hosted) user the `--serverTimeout=1800` for this (I heard from some they even needed a larger number then that)

Those settings are in my main system, so I don't have to download Chrome on every install. Similar setup will work for CI/CD too, just make sure the executable is available _before_ you start Scully
```sh
export CHROMIUM_PATH=/home/sander/.apps/chrome-linux/chrome
export SCULLY_PUPPETEER_EXECUTABLE_PATH=/home/sander/.apps/chrome-linux/chrome
export PUPPETEER_EXECUTABLE_PATH=/home/sander/.apps/chrome-linux/chrome
export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
```
